### PR TITLE
ITNFA-14: Persisting the national fingerprint identifier as a part of the search

### PR DESCRIPTION
**Pull request by oskarhinc**
… the search

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java [hunk 1]
```diff
@@ -74,6 +86,29 @@ public void fetchFromNational(BiometricSubject nationalBiometricSubject, M2SysCa
         registerLocally(nationalBiometricSubject, fingerScan);
     }
 
+    @Override
+    public void synchronizeFingerprints(M2SysCaptureResponse fingerScan, FingerScanStatus fingerScanStatus) {
+        if (fingerScanStatus.isRegisteredNationally()) {
+            if (patientHelper.findByNationalFpId(fingerScanStatus.getNationalBiometricSubject().getSubjectId()) == null) {
+                attachNationalIdToThePatient(fingerScanStatus.getLocalBiometricSubject().getSubjectId());
```
Comment (dserkowski) [inline api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java]: Why getLocalBiometricSubject? I think there should be getNationalBiometricSubject.

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java [hunk 2]
```diff
@@ -74,6 +86,29 @@ public void fetchFromNational(BiometricSubject nationalBiometricSubject, M2SysCa
...
+            }
+        } else {
+            registerNationally(fingerScanStatus.getLocalBiometricSubject(), fingerScan);
+        }
+    }
+
+    private void attachNationalIdToThePatient(String localId) {
```
Comment (dserkowski) [inline api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java]: I think that you should pass Patient and nationalId here or name this method more appropriately. It is a bit confusing and difficult to understand.

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java [hunk 3]
```diff
@@ -74,6 +86,29 @@ public void fetchFromNational(BiometricSubject nationalBiometricSubject, M2SysCa
...
+        Patient patient = patientHelper.findByLocalFpId(localId);
+
+        PatientIdentifierType patientIdentifierType = patientService.getPatientIdentifierTypeByUuid(
+                properties.getGlobalProperty(RegistrationCoreConstants.GP_BIOMETRICS_NATIONAL_PERSON_IDENTIFIER_TYPE_UUID));
+        Location location = locationService.getDefaultLocation();
```
Comment (dserkowski) [inline api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java]: I think it is ok, but I have noticed that in some places (in i.e. registrationcore) the location is set null. @pgesek, what do you think about it?

**Review by dserkowski — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/util/impl/PatientHelperImpl.java [hunk 4]
```diff
@@ -44,8 +46,13 @@ private Patient findByIdType(String subjectId, String idTypeProp) {
             if (idType == null) {
                 LOGGER.warn("Identifier type defined by prop {} is missing: {}", idTypeProp, identifierUuid);
             } else {
+                //Currently method getPatients() do not take into consideration the identifierTypes,
```
Comment (pawelcieszko) [inline api/src/main/java/org/openmrs/module/m2sysbiometrics/util/impl/PatientHelperImpl.java]: do -> doesn't

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java (hunk 2 above)
Comment (pgesek) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java]: Yes, I don't get why are we saving the localID as the national ID

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java (hunk 3 above)
Comment (pgesek) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java]: If this is the national id we can skip the id - but I'm not sure which id this s

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java (hunk 1 above)
Comment (oskarhinc) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java]: You're right

File: api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java (hunk 3 above)
Comment (oskarhinc) [reply api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java]: Location cannot be null in this case. The database won't accept that.

**Comment by dserkowski** [comment]
LGTM

**Review by dserkowski — APPROVED** [review]

**Pull request merged**